### PR TITLE
support spark-version project

### PR DESCRIPTION
**Pull request by hhbyyh**
add Spark-version project (similar to the one in BigDL) to support different component in Spark 1.6+ and Spark 2.X, E.g. Vector and UDF.

### CODE REVIEW

**Comment by hhbyyh** [comment]
cc @zhichao-li @glorysdj

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/902/  passed.

**Comment by hhbyyh** [comment]
@glorysdj Can you please help take a look? Thanks
also cc @jason-dai

**Comment by hhbyyh** [comment]
cc @glorysdj updated with master code. Can you please help review this? Thanks

File: pom.xml [hunk 1]
```diff
@@ -135,7 +143,7 @@
             </plugins>
         </pluginManagement>
         <plugins>
-	       <plugin>
+            <plugin>
```
Comment (glorysdj) [inline pom.xml]: should keep the format

**Comment by glorysdj** [comment]
how should this tests? via jenkins pr validation? or we should copy a Zoo-Release job and modify it to do profiles tests?

**Comment by hhbyyh** [comment]
Close this for now as ML Vector is supported. Thanks for the review @glorysdj

**Pull request closed without merging**